sql/core.go

```diff
@@ -370,14 +371,56 @@ type SystemVariableRegistry interface {
 	NewSessionMap() map[string]SystemVarValue
 	// GetGlobal returns the global value of the system variable with the given name
 	GetGlobal(name string) (SystemVariable, interface{}, bool)
+	// GetResetVal returns the default value of the system variable with the given name.
+	// This is not the current GLOBAL value. It's used for postgres config parameters only.
+	GetResetVal(name string) (SystemVariable, interface{}, bool)
 	// SetGlobal sets the global value of the system variable with the given name
 	SetGlobal(name string, val interface{}) error
 	// GetAllGlobalVariables returns a copy of all global variable values.
 	GetAllGlobalVariables() map[string]interface{}
 }
 
-// SystemVariable represents a system variable.
-type SystemVariable struct {
+// SystemVariable is used to handle mysql system variables and postgres configuration variables interchangeably.
+// TODO: mysql sv and postgres cp should not be mixed.
+type SystemVariable interface {
+	// GetName returns the name of the sv. Case-sensitive.
+	GetName() string
+	// SetName sets the name of the sv.
+	// TODO: it should not happen, but we set the name to lower-case string?
+	SetName(string)
```

Comment:
Is this allowed? Renaming a variable would change the variable, so it seems like it shouldn't be allowed. Not sure where this is used. If it is necessary for something, then we should name it `ForceChangeName` or something similar, with an appropriate comment mentioning that it shouldn't be used or the cases in which it should only be used.

Reply:
I think this is not allowed, but we use it internally in only one place, which is in `AddSystemVariables` method. We use `SetName` method to make sure the variable name is lower-cased before adding the variable to the global `SystemVariables` map. 
I personally think this is not necessary/incorrect, as this will update couple of the Postgres parameter names to be updated into incorrect names (e.g. `TimeZone` and `DateStyle`, which are case-sensitive) since we use `AddSystemVariables` method to add postgres parameters to the `SystemVariables` map.

Reply:
In that case, we should add some other way to enforce that the names are lowercase for the GMS/Dolt path, but isn’t run from Doltgres. We could even check if it’s a MySQL struct, since Doltgres won’t use those, then we can remove the function.

Reply:
I'm removing them as I think use case-insensitive name for SV map key for storing and use the case-sensitive defined name for displaying the variable.